app/services/GetCalculationDetailsService.scala

```diff
@@ -28,15 +28,26 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class GetCalculationDetailsService @Inject()(calculationDetailsConnectorLegacy: CalculationDetailsConnectorLegacy,
                                              calculationDetailsConnector: CalculationDetailsConnector,
-                                             listCalculationDetailsConnector: GetCalculationListConnector) (implicit ec: ExecutionContext) {
+                                             listCalculationDetailsConnector: GetCalculationListConnector,
+                                             listCalculationDetailsConnectorLegacy: GetCalculationListConnectorLegacy) (implicit ec: ExecutionContext) {
 
   def getCalculationDetails(nino: String, taxYear: Option[String])(implicit hc: HeaderCarrier):Future[CalculationDetailResponse] = {
-    listCalculationDetailsConnector.calcList(nino, taxYear).flatMap {
-      case Right(listOfCalculationDetails) if(listOfCalculationDetails.isEmpty) =>
-        Future.successful(Left(DesErrorModel(NO_CONTENT, DesErrorBodyModel.parsingError)))
-      case Right(listOfCalculationDetails) =>
-        getCalculationDetailsByCalcId(nino, listOfCalculationDetails.head.calculationId, taxYear)
-      case Left(desError) => Future.successful(Left(desError))
+    if (taxYear.isDefined && taxYear.get == "2024") {
```

Comment:
this might need to be for any taxYear going forward from 2023-24, not specifically for a single year

Reply:
@avelx This API is specifically for 23-24 and we do not have any confirmation beyond that so we are asked to use this API only for 23-24

Reply:
This API is specifically for 2023-24. 

From the description text in the API schema.
`'API#1896 Get Calculation List (TY 23-24)'`

`23-24` is also the only permissible value for the tax year parameter.